### DISCUSSION TITLE
Is it possible to use System.Text.Json with NewLineDelimitedMessageHandler?

### DISCUSSION THREAD

**Discussion by mmoraga** [microsoft/vs-streamjsonrpc#959] [Q&A] [answered]
using vs-streamjsonrpc v.2.16.36

I'm trying to replace use of Newtonsoft.Json with System.Text.Json, but supplying SystemTextJsonFormatter to the NewLineDelimitedMessageHandler just throws the NotSupportedException

```
var messageHandler = new NewLineDelimitedMessageHandler(stream, stream, new SystemTextJsonFormatter());
```

I see this comment in the source code of NewLineDelimitedMessageHandler:
```
if (this.Formatter is IJsonRpcFormatterTracingCallbacks)
        {
            // Such a formatter requires that we make their own written bytes available back to them for logging purposes.
            // We haven't implemented support for such, and the JsonMessageFormatter doesn't need it, so no need to.
            throw new NotSupportedException("Formatters that implement " + nameof(IJsonRpcFormatterTracingCallbacks) + " are not supported. Try using " + nameof(JsonMessageFormatter) + ".");
        }
```

Are you planning to add this support in the future? Or is there some sort of workaround? I don't understand why HeaderDelimitedMessageHandler would be supported but not NewLineDelimitedMessageHandler.

**Comment by AArnott** [comment, marked as answer]
We had no plans. But it looks easy enough to enhance. https://github.com/microsoft/vs-streamjsonrpc/pull/962

**Reply by mmoraga** [reply]
Thank you!
